concordium-contracts-common/src/schema.rs

```diff
@@ -755,26 +758,54 @@ impl Deserial for FunctionV2 {
     fn deserial<R: Read>(source: &mut R) -> ParseResult<Self> {
         let idx = source.read_u8()?;
         match idx {
-            0 => Ok(FunctionV2::Param(source.get()?)),
-            1 => Ok(FunctionV2::Rv(source.get()?)),
-            2 => Ok(FunctionV2::ParamRv {
+            // parameter
+            0 => Ok(FunctionV2 {
                 parameter:    source.get()?,
+                return_value: None,
+                error:        None,
+            }),
+            // return_value
+            1 => Ok(FunctionV2 {
+                parameter:    None,
                 return_value: source.get()?,
+                error:        None,
             }),
-            3 => Ok(FunctionV2::Error(source.get()?)),
-            4 => Ok(FunctionV2::ParamError {
-                parameter: source.get()?,
-                error:     source.get()?,
+            // parameter + return_value
+            2 => Ok(FunctionV2 {
+                parameter:    source.get()?,
+                return_value: source.get()?,
+                error:        None,
+            }),
+            // error
+            3 => Ok(FunctionV2 {
+                parameter:    None,
+                return_value: None,
+                error:        source.get()?,
             }),
-            5 => Ok(FunctionV2::RvError {
+            // parameter + error
+            4 => Ok(FunctionV2 {
+                parameter:    source.get()?,
+                return_value: None,
+                error:        source.get()?,
+            }),
+            // return_value + error
+            5 => Ok(FunctionV2 {
+                parameter:    None,
                 return_value: source.get()?,
                 error:        source.get()?,
             }),
-            6 => Ok(FunctionV2::ParamRvError {
+            // parameter + return_value + error
+            6 => Ok(FunctionV2 {
                 parameter:    source.get()?,
                 return_value: source.get()?,
                 error:        source.get()?,
             }),
+            // no schema
+            7 => Ok(FunctionV2 {
+                parameter:    None,
+                return_value: None,
+                error:        None,
+            }),
             _ => Err(ParseError::default()),
         }
     }
```

Comment:
I think this has the same issue as Serial, in that `source.get()` you are actually deserializing `Option<Type>`. But you want `Type` since the tag already tells you the option. I think you want this

```rust

impl Deserial for FunctionV2 {
    fn deserial<R: Read>(source: &mut R) -> ParseResult<Self> {
        let idx = source.read_u8()?;
        let mut r = FunctionV2 {
            parameter:    None,
            return_value: None,
            error:        None,
        };
        if idx > 7 {
            return Err(ParseError::default());
        }
        if matches!(idx, 0 | 2 | 4 | 6) {
            let _ = r.parameter.insert(source.get()?);
        }
        if matches!(idx, 1 | 2 | 5 | 6) {
            let _ = r.return_value.insert(source.get()?);
        }
        if matches!(idx, 3 | 4 | 5 | 6) {
            let _ = r.error.insert(source.get()?);
        }
        Ok(r)
    }
}
```